app/models/host/managed.rb

```diff
@@ -320,10 +320,10 @@ class Jail < ::Safemode::Jail
                         :if => proc { |host| host.managed && host.disk.empty? && !Foreman.in_rake? && !host.image_build? && host.build? }
   validates :provision_method, :inclusion => {:in => proc { provision_methods }, :message => N_('is unknown')}, :if => proc { |host| host.managed? }
   validates :medium_id, :presence => true,
-                        :if => proc { |host| host.validate_media? }
+                        :if => proc { |host| host.managed? && host.validate_media? }
   validates :medium_id, :inclusion => {:in => proc { |host| host.operatingsystem.medium_ids },
                                        :message => N_('must belong to host\'s operating system')},
-                        :if => proc { |host| host.operatingsystem && host.medium }
+                        :if => proc { |host| host.managed? && host.operatingsystem && host.medium }
```

Comment:
Is this really correct? If it's unmanaged, why does it even have a medium in the first place? And if it doesn't have a medium, the check isn't blocking.

IMHO that's really the root cause: somehow the host is getting a medium assigned that's invalid for the operating system. We should not do that.

Perhaps this is a bug in the inheritance from a hostgroup?

Reply:
A host was registered using HostRegistartion. During this process, a Hostgroup was used and assigned to the host. 
For this Hostgroup the Installation Media was configured. 

The host is using the subscription-manager to register. This will create a new OS in foreman if the OS doesn't currently exist. This new OS doesn’t have the installation media assigned (because its a new OS).

Summary:
- The new created Operatingsystem like AlmaLinux9 doesn't have a Installation Media.
- The used Hostgroup has different Operatingsystems and is using a InstallationMedia which is valid for the Operatingsystem used in this Hostgroup.

The POST request during the HostRegistration process which assign Hostgroup to the Host and should render the host_init_config Template will fail with:

```
The system has been registered with ID: 21312a8f-1b6e-4302-ae9e-1a64b643c6f8
The registered system name is: myra-bergh.master.dev.atix
ERROR: Validation failed: Medium must belong to host's operating system
```

Notice:
- if the Hostgroup doesn't have a Operatingsystem OR Installation Media > everything works
- If the Operatingsystem of the Host, is already configured on foreman, including valid Installatin Media -> everything works